azext_hanaonazure/custom.py

```diff
@@ -92,17 +96,81 @@ def list_sapmonitor(client):
 def show_sapmonitor(client, resource_group_name, monitor_name):
     return client.get(resource_group_name, monitor_name)
 
-def create_sapmonitor(client, resource_group_name, monitor_name, region, hana_subnet, hana_hostname, hana_db_sql_port, hana_db_username, hana_db_password, hana_db_name):
+def create_sapmonitor(
+        cmd,
+        client,
+        resource_group_name,
+        monitor_name,
+        region,
+        hana_subnet,
+        hana_hostname,
+        hana_db_sql_port,
+        hana_db_username,
+        hana_db_name,
+        hana_db_password=None,
+        hana_db_password_key_vault_url=None,
+        key_vault_id=None):
+
     monitoring_details = {
         "location": region,
         "hanaSubnet": hana_subnet,
         "hanaHostname": hana_hostname,
         "hanaDbName": hana_db_name,
         "hanaDbSqlPort": hana_db_sql_port,
-        "hanaDbUsername": hana_db_username,
-        "hanaDbPassword": hana_db_password
+        "hanaDbUsername": hana_db_username
     }
+
+    if hana_db_password is not None:
+        # Password was passed in
+        monitoring_details.update({"hanaDbPassword": hana_db_password})
+    elif hana_db_password_key_vault_url is not None and key_vault_id is not None:
+        # Keyvault URL was passed in
+        from ._client_factory import (_msi_client_factory, _keyvault_client_factory)
+        from azure.mgmt.keyvault.v2018_02_14.models import (VaultAccessPolicyProperties, AccessPolicyEntry, Permissions, SecretPermissions)
+
+        # Create MSI
+        msi_client = _msi_client_factory(cmd.cli_ctx)
+        arm_resource_id = arm_resource_id_format.format(msi_client.config.subscription_id, resource_group_name, monitor_name)
+        sapmon_id = 'sapmon{}'.format(fnv32a(arm_resource_id))
```

Comment:
Should be: `sapmon_id = 'sapmon-msi-{}'.format(fnv32a(arm_resource_id))`
As per [naming convention](https://microsoft.sharepoint.com/teams/AzureDedicatedEngineeringTeam/_layouts/OneNote.aspx?id=%2Fteams%2FAzureDedicatedEngineeringTeam%2FSiteAssets%2FAzure%20Dedicated%20Engineering%20Team%20Notebook&wd=target%28SAP%20HANA.one%7CFB7D0654-1276-104F-AF53-BA55FC4DA39E%2FDesign%7CE2184110-0189-3940-AC33-F400641A036D%2F%29)